### PR TITLE
docs: ensure up-to-date

### PR DESCRIPTION
**Pull request by straker**
The [API docs for standards](https://github.com/dequelabs/axe-core/pull/2417/files) has a few additional things I didn't pull over. Also, the formatting of newlines was a bit odd so I fixed some of that too where I saw it.

Closes issues: #2277, #1936, #2323, and #840

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -836,25 +854,6 @@ axe.commons.dom.getRootNode(node);
 
 The top-level document or shadow DOM document fragment
 
-#### axe.commons.dom.findUp
```
Comment (straker) [inline doc/API.md:839]: `findUp` was deprecated.

File: doc/API.md [hunk 2]
```diff
@@ -54,19 +53,7 @@ The axe API can be used as part of a broader process that is performed on many,
 
 ### Overview
 
-The axe APIs are provided in the javascript file axe.js. It must be included in the web page under test. Parameters are sent as javascript function parameters. Results are returned in JSON format.
-
-### Required Globals
```
Comment (straker) [inline doc/API.md:59]: We no longer require globals.

File: doc/developer-guide.md [hunk 3]
```diff
@@ -31,24 +31,24 @@ Axe 3.0 supports open Shadow DOM: see our virtual DOM APIs and test utilities fo
 ### Environment Pre-requisites
 
 1. You must have NodeJS installed.
-2. Grunt must be installed globally. `npm install -g grunt-cli` (You may need to do this as `sudo npm install -g grunt-cli`)
```
Comment (straker) [inline doc/developer-guide.md:34]: We don't need `grunt` globally to run axe

File: doc/API.md [hunk 4]
```diff
@@ -400,7 +412,7 @@ Additionally, there are a number or properties that allow configuration of diffe
 | `rules`            | n/a     | Enable or disable rules using the `enabled` property                                                                                    |
 | `reporter`         | `v1`    | Which reporter to use (see [Configuration](#api-name-axeconfigure))                                                                     |
 | `resultTypes`      | n/a     | Limit which result types are processed and aggregated                                                                                   |
-| `selector`         | `true`  | Return CSS selector for elements, optimised for readability                                                                             |
+| `selectors`        | `true`  | Return CSS selector for elements, optimised for readability                                                                             |
```
Comment (WilcoFiers) [inline doc/API.md:415]: oops!

File: doc/developer-guide.md [hunk 5]
```diff
@@ -187,73 +189,72 @@ A CheckResult has the following properties:
...
+  attr: <Function>,
+  hasAttr: <Function>,
+  props: <Object>,
 }
 ```
 
 - A virtualNode is an object containing an HTML DOM element (`actualNode`).
-- Children contains an array of child virtualNodes.
+- Children contains an array of child VirtualNodes.
+- Parent is the VirutalNode parent
```
Comment (WilcoFiers) [inline doc/developer-guide.md:240]:
```suggestion
- Parent is the VirtualNode parent
```

**Pull request merged**